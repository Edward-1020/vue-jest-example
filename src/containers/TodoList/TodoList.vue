<template>
  <div>
    <h1>TodoList</h1>
  </div>
</template>

<script type="text/ecmascript-6">
export default {
  name: 'TodoList'
}
</script>

<style scoped lang="scss" rel="stylesheet/scss">
</style>
