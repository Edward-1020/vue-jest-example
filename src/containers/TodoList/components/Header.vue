<template>
  <div class="header">
    <div class="header-content">
      TodoList:
      <input
        placeholder="add todoItem"
        class="header-input"
        data-test="input"
        v-model="inputValue"
        @keyup.enter="addTodoItem"
        type="text"
      />
    </div>
  </div>
</template>

<script type="text/ecmascript-6">
export default {
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="scss" rel="stylesheet/scss">
  .header {
    line-height: 60px;
    background: #333;
  }
  .header-content {
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
  }
  .header-input {
    float: right;
    width: 360px;
    margin-top: 16px;
    line-height: 24px;
    outline: none;
    color: #333;
  }
</style>
