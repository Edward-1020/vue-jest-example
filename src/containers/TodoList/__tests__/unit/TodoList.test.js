import { shallowMount } from '@vue/test-utils'
import TodoList from '../../TodoList.vue'
import Header from '../../components/Header.vue'

it('Header 样式发生改变，做提示', () => {
  const wrapper = shallowMount(Header);
  expect(wrapper).toMatchSnapshot();
})

it('TodoItem 监听到 Header的add事件时，会增加一个内容', () => {
  const content = 'dell lee';
  const wrapper = shallowMount(TodoList)
  const header = wrapper.find(Header);
  header.vm.$emit('add', content)
  const undoList = wrapper.vm.$data.undoList;
  expect(undoList).toEqual([content]);
})
