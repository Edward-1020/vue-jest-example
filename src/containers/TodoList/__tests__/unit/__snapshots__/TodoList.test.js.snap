// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList:
    <input placeholder="add todoItem" data-test="input" type="text" class="header-input"></div>
</div>
`;
